pennylane/ops/op_math/adjoint.py

```diff
@@ -273,10 +273,11 @@ def sparse_matrix(self, wire_order=None, format="csr"):
     def has_decomposition(self):
         return self.base.has_adjoint or self.base.has_decomposition
 
-    def decomposition(self):
-        if self.base.has_adjoint:
-            return [self.base.adjoint()]
-        base_decomp = self.base.decomposition()
+    @staticmethod
+    def compute_decomposition(*args, wires, base, **kwargs):
```

Comment:
So to explain our `compute_representation` versus `representation` separation.  `compute_representation` is more of a helper method that operators can implement instead, but the `op.representation` method is what things other things should actually rely on being defined.

Basically, if `compute_decomposition` is defined, `decomposition` also needs to be defined.  BUT, `decomposition` can be defined without `compute_decomposition` being defined.

Sometimes its easier just to define `decomposition`.  Like for this, `*args` and `**kwargs` have no impact on the decomposition, so we are unpacking and hauling around some data just for it not to get used.

Also, the only hyperparameter for `Adjoint` should be `base`, so we don't need `**kwargs`.

Reply:
the main reason I made this change was just for consistency with the rest of the code base. Every Operator has the default `decomposition` implementation (which just calls `compute_decomposition`), and I figured why not stick to that? I realized because I was searching for "def decomposition(" and "def compute_decomposition(". They both get the job done, I'd just prefer we're consistent. Though tbh, I'm not sure what the value of the static method is. It's usually for caching purposes, right? Like with `matrix`?

Reply:
@timmysilv for background, there were two reasons we decided to have both:

- We had feedback from users that they wanted/were relying on static methods to compute representations of operators, and it would be valuable to retain that in our new operator API (the above is the 'new' operator API!)

- For numerical representations, static or class methods would allow us to cache expensive computations. E.g., in large circuits, RX(2pi) may be called thousands of times. We did some benchmarking, and identified that for simulators that rely on `op.matrix`, caching `op.compute_matrix` _across_ all op instantiations would lead to an order of magnitude speed up.

Tagging @mariaschuld in case there were other reasons I have forgotten

Reply:
In other words, overriding the static `compute_decomposition` _is_ preferred over overriding `decomposition`?

Reply:
In our original design, yeah - I recall the instruction was 'override `compute_*` where possible; if difficult due to some constraint, you still have the option to override `*` directly.

Reply:
And in this situation, I would say say its unwieldy to override `compute_*` due to complexity of the call signature. 

We could potentially override *both* `compute_decomposition` and `decomposition`, and give it the call signature `compute_decomposition(base: Operator)` without args and kwargs, but I'm not a fan of the `def compute_decomposition(*args, wires, base, **kwargs):` call signature.

We could also consider moving this debate to a new PR where we focus on conventions for `compute_*` methods for operator arithmetic.

Reply:
I'll remove it from this PR since it's pretty unrelated to the goal here.